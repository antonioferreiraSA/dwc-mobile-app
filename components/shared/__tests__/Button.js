import React from 'react';
import { View, Text } from 'react-native';
import { render, fireEvent } from '@testing-library/react-native';
import Button from '../Button';

describe('<Button/>', () => {
  test('render the button', async () => {
    const { toJSON } = render(<Button title="Button" />);

    expect(toJSON()).toMatchSnapshot();
  });

  test('render the button with style', async () => {
    const styles = { backgroundColor: 'black' };
    const { toJSON } = render(
      <Button title="Button" style={styles} textStyle={styles} />
    );

    expect(toJSON()).toMatchSnapshot();
  });

  test('render the button with an icon', async () => {
    const { toJSON } = render(
      <Button
        title="Button"
        icon={
          <View>
            <Text>Icon</Text>
          </View>
        }
      />
    );

    expect(toJSON()).toMatchSnapshot();
  });

  test('render the button and press it', async () => {
    const mockPress = jest.fn();
    const { getByText } = render(<Button title="Button" onPress={mockPress} />);

    fireEvent.press(getByText('Button'));

    expect(mockPress).toBeCalled();
  });
});
