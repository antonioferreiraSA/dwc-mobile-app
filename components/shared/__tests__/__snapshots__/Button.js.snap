// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> render the button 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#313d48",
        "borderRadius": 30,
        "justifyContent": "center",
        "minHeight": 60,
        "paddingHorizontal": 16,
        "paddingVertical": 12,
      },
      Object {},
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      selectable={true}
      style={
        Array [
          Object {
            "color": "#cbd2d6",
            "fontFamily": "NunitoSans-Regular",
            "fontSize": 18,
            "lineHeight": 24,
            "textAlign": "auto",
          },
          Array [
            Object {
              "color": "#fff",
              "textAlign": "center",
            },
            Object {},
          ],
        ]
      }
    >
      Button
    </Text>
  </View>
</View>
`;

exports[`<Button/> render the button with an icon 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#313d48",
        "borderRadius": 30,
        "justifyContent": "center",
        "minHeight": 60,
        "paddingHorizontal": 16,
        "paddingVertical": 12,
      },
      Object {},
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "marginRight": 10,
        }
      }
    >
      <View>
        <Text>
          Icon
        </Text>
      </View>
    </View>
    <Text
      selectable={true}
      style={
        Array [
          Object {
            "color": "#cbd2d6",
            "fontFamily": "NunitoSans-Regular",
            "fontSize": 18,
            "lineHeight": 24,
            "textAlign": "auto",
          },
          Array [
            Object {
              "color": "#fff",
              "textAlign": "center",
            },
            Object {},
          ],
        ]
      }
    >
      Button
    </Text>
  </View>
</View>
`;

exports[`<Button/> render the button with style 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#313d48",
        "borderRadius": 30,
        "justifyContent": "center",
        "minHeight": 60,
        "paddingHorizontal": 16,
        "paddingVertical": 12,
      },
      Object {
        "backgroundColor": "black",
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      selectable={true}
      style={
        Array [
          Object {
            "color": "#cbd2d6",
            "fontFamily": "NunitoSans-Regular",
            "fontSize": 18,
            "lineHeight": 24,
            "textAlign": "auto",
          },
          Array [
            Object {
              "color": "#fff",
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "black",
            },
          ],
        ]
      }
    >
      Button
    </Text>
  </View>
</View>
`;
